src/utils/helpers/trusted-algorithms.ts

```diff
@@ -0,0 +1,73 @@
+import {
+  Asset,
+  FileInfo,
+  ProviderInstance,
+  PublisherTrustedAlgorithm,
+  getHash
+} from '@oceanprotocol/lib'
+import { Nautilus } from '../../Nautilus'
+import { getAsset } from '../aquarius'
+
+// TODO replace hardcoded service index 0 with service id once supported by the stack
+export async function getPublisherTrustedAlgorithms(
+  dids: string[],
+  nautilus: Nautilus
```

Comment:
suggestion: as this is a utility function, I prefer for it to work without relying on any repo class instances. In this case I think `nautilus` is only used to retrieve the `metadataCacheUri` from the config. To break responsibilities a bit better here, imo the helper should simply work with passing a `metadataCacheUri` parameter.

```ts
export async function getPublisherTrustedAlgorithms(
  dids: string[],
  metadataCacheUri: string
) {
  const assetPromises = dids.map((did) =>
    getAsset(metadataCacheUri, did)
  )
// ... 
}
```